app/controllers/matches_controller.rb

```diff
@@ -13,22 +13,38 @@ def show
   # PATCH/PUT /matches/1
   def update
     new_state = match_params['state']
-    if @match.update(match_params)
-      if new_state == 'finished'
-        result = PopulateMatchBelowAndSave.call(match: @match) unless @match.group_match?
-        unless result.success?
-          render json: { error: 'Moving Team one stage down failed' }, status: :unprocessable_entity
-          return
-        end
+
+    Match.transaction do
+      if @match.update(match_params)
+        return handle_match_end if new_state == 'finished'
```

Comment:
what is the return doing here?

Reply:
At some point in development it was necessary there.. removed it now